### ISSUE TITLE
jsx-uses-vars is confused by shadowed variables.

### ISSUE THREAD

**Issue by Wilfred** [yannickcr/eslint-plugin-react#799]
Given the following code:

```js
import React from 'react';
import {FooPicker} from '../FooPicker';

function myTest() {
  const FooPicker = require('../FooPicker').default;
  return <FooPicker />;
}

myTest();
```

and the following `.eslintrc.json`:

```json
{
    "env": {
        "node": true
    },
    "parserOptions": {
        "ecmaVersion": 6,
        "ecmaFeatures": {
            "jsx": true,
            "impliedStrict": true
        },
        "sourceType": "module"
    },
    "rules": {
        "no-unused-vars": "warn",
        "react/jsx-uses-vars": "warn",
        "react/jsx-uses-react": "warn"
    },
    "plugins": [
        "react"
    ]
}
```

No warnings are generated. We should warn on the `import {FooPicker}`, as it's unused.

Possibly related: #716.

**Comment by Wilfred** [comment]
Thanks! :)
